internal/core/attribute.cpp

```diff
@@ -30,33 +30,42 @@ Horizon::Window Horizon::at(std::size_t i) const noexcept (false) {
     };
 }
 
+void AttributeMap::AttributeComputeParam::update(
+    InputIt begin,
+    InputIt end,
+    std::size_t size,
+    std::size_t reference_index
+) noexcept (true) {
+    this->begin = begin;
+    this->end = end;
+    this->size = size;
```

Comment:
> but is there a benefit of getting rid of size parameter and using std::distance instead?
Less variables floating around

Your call tho

Reply:
I've tried removing it and I do not like it. I think if you provide parameters as an object class, it irritates me less than when parameters were provided on construction and were visible in signature.

Now it is hidden and can be just considered calculation, like storing something in a variable because it is nicer than calculating the same value in every loop iteration.

So leaving as is.